### DISCUSSION TITLE
[filter] Time Grain control for dashboard user

### DISCUSSION THREAD

**Discussion by kamalkeshavani-aiinside** [apache/superset#18497] [Ideas]
**Is your feature request related to a problem? Please describe.**
There are cases when we want the dashboard users/viewers to be able to change Time Grain of a time-series chart. Currently this is possible from 'Show SQL Granularity Dropdown' option in Filter chart.
Here, we would like to be able to control the Time Grains available to the dashboard end user. With current implementation, all the available granularity options will be made available to the end user, which can be unnecessary as well as confusing to some users.

**Describe the solution you'd like**
When we enable 'Show SQL Granularity Dropdown' option in Filter chart, we should be able to select the granularity options  available/visible to the dashboard user. Default behavior will be all available as it is now.

**Describe alternatives you've considered**

**Additional context**

**Comment by junlincc** [comment]
> Here, we would like to be able to control the Time Grains available to the dashboard end user. With current implementation, all the available granularity options will be made available to the end user, which can be unnecessary as well as confusing to some users.

Totally agree with your pain points. We rushed to get this feature out earlier this year without much design thinking. 
Not sure if we wanna add a configuration or simply make showing time grain option "smarter". If your org has design resource, we would love to collaborate on it. @kamalkeshavani-aiinside 

cc @villebro @zhaoyongjie

**Comment by kamalkeshavani-aiinside** [comment]
@junlincc Thanks for taking up the request. There is no other special resource available in my org available to contribute here now, I will try to support as much as I can.
